### ISSUE TITLE
Incompatible return type not detected when overwriting 

### ISSUE THREAD

**Issue by iknow125** [microsoft/pyright#4997]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
pyright does not detect a wrong return type when overwriting a method. This is especially critical when using abstract classes since some of the children couldn't  match the contract/interface.

**To Reproduce**
```bash
❯ python -m mypy testing/test.py
testing/test.py:7: error: Return type "str" of "something" incompatible with return type "None" in supertype "Parent"  [override]
Found 1 error in 1 file (checked 1 source file)
```
---
```bash
❯ python -m pyright testing/test.py
WARNING: there is a new pyright version available (v1.1.303 -> v1.1.304).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

No configuration file found.
No pyproject.toml file found.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
pyright 1.1.303
0 errors, 0 warnings, 0 informations 
Completed in 0.84sec
```

**Expected behavior**
There should be some error that the return type is not correct 

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
class Parent:
    def something(self) -> None:
        pass


class Child(Parent):
    def something(self) -> str:
        return "Hi"

```

If your code relies on symbols that are imported from a third-party library, include the associated import statements and specify which versions of those libraries you have installed.

**Comment by hmc-cs-mdrissi** [comment]
reportIncompatibleMethodOverride is the rule you are looking for. It is off by default. It is true in strict mode and you can turn it on yourself. So pyright does support this.

The difference is exact rules that are on/off in basic mode vs strict mode differ between mypy and pyright. Opposite example of mypy being less strict in basic mode is mypy defaults to not type checking body of functions that are not type annotated for the definition.

**Comment by iknow125** [comment]
Thank you for the reply :)
